src/Microsoft.Azure.SignalR/ProductInfo.cs

```diff
@@ -8,16 +8,27 @@ namespace Microsoft.Azure.SignalR
 {
     internal static class ProductInfo
     {
+        private const int MaxLength = 128;
+
         public static string GetProductInfo()
         {
-            var assembly = Assembly.GetCallingAssembly();
+            return GetProductInfo(Assembly.GetCallingAssembly());
+        }
+
+        public static string GetProductInfo(Assembly assembly)
+        {
             var packageId = assembly.GetName().Name;
             var version = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
             var runtime = RuntimeInformation.FrameworkDescription?.Trim();
             var operatingSystem = RuntimeInformation.OSDescription?.Trim();
             var processorArchitecture = RuntimeInformation.ProcessArchitecture.ToString().Trim();
-            
-            return $"{packageId}/{version} ({runtime}; {operatingSystem}; {processorArchitecture})";
+            var packageInfo = $"{packageId}/{version}";
+            return $"{TruncateString(packageInfo)} ({runtime}; {operatingSystem}; {processorArchitecture})";
```

Comment:
>TruncateString [](start = 22, length = 14)

which one could probabaly exceed max?

Reply:
append `...` to indicate it is truncated?

---
In reply to: [279599840](https://github.com/Azure/azure-signalr/pull/513#discussion_r279599840) [](ancestors = 279599840)